crates/sui-source-validation/src/tests.rs

```diff
@@ -30,23 +33,85 @@ async fn successful_verification() -> anyhow::Result<()> {
         publish_package(context, sender, b_src).await
     };
 
-    let a_pkg = {
+    let (a_pkg, a_ref) = {
         let fixtures = tempfile::tempdir()?;
         let b_id = b_ref.0.into();
         copy_package(&fixtures, "b", [("b", b_id)]).await?;
         let a_src = copy_package(&fixtures, "a", [("a", SuiAddress::ZERO), ("b", b_id)]).await?;
-        compile_package(a_src)
+        (
+            compile_package(a_src.clone()),
+            publish_package(context, sender, a_src).await,
+        )
     };
     let client = context.get_client().await?;
     let verifier = BytecodeSourceVerifier::new(client.read_api(), false);
+    let a_addr: SuiAddress = a_ref.0.into();
+
+    // Skip deps and root
+    verifier
+        .verify_package(&a_pkg.package, false, None)
+        .await
+        .unwrap();
+    sleep(Duration::from_millis(1000)).await;
+
+    // Verify deps but skip root
+    verifier
+        .verify_package(&a_pkg.package, true, None)
+        .await
+        .unwrap();
+    sleep(Duration::from_millis(1000)).await;
+
+    // Skip deps but verify root
     verifier
-        .verify_deployed_dependencies(&a_pkg.package)
+        .verify_package(&a_pkg.package, false, Some(a_addr.into()))
```

Comment:
Related to the comments about simplifying how root compiled unit verification works, you'll need to tweak the test so that you compile and publish `a` with address zero, then take its published address and create a new copy of the fixture with the address set to how it is when published.